src/platform/silabs/wifi/SiWx/WifiInterfaceImpl.cpp

```diff
@@ -698,7 +698,20 @@ sl_status_t WifiInterfaceImpl::JoinWifiNetwork(void)
 
     status = sl_net_up(SL_NET_WIFI_CLIENT_INTERFACE, SL_NET_DEFAULT_WIFI_CLIENT_PROFILE_ID);
 
-    if (status == SL_STATUS_OK || status == SL_STATUS_IN_PROGRESS)
+    if (!(wfx_rsi.dev_state.Has(WifiInterface::WifiState::kStationConnecting)))
+    {
+        // TODO: Remove this check once the sl_net_up is fixed, sl_net_up is not completely synchronous
```

Comment:
These are marked as SL-TEMP.  We should split the PRs imo to have the correct changes marked SL-UP and SL-TEMP.

Reply:
Since the fix will be in v4.0.0, this needs to be brought to the CSA. Will discuss with Rohan as well about it.